From a15cc222997c1def4f1b38db7a847b95238cebce Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Wed, 5 Jul 2023 14:58:21 +0200
Subject: [PATCH 0398/1695] drivers: stm32_rng: put max noise freq in compat
 data

This avoids having config flags in the driver. The max noise clock
frequency is changed as an entry in RNG compta data structure.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 429a01e666fdebf2a9fec072eb7caaf3dbd4caf0 drivers: stm32_rng: put max noise freq in compat data]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I21404426f45abdd1f7557d4fb3694a9df9b193da
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/304349
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_rng.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index 80d363711..f6111e08f 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -69,9 +69,8 @@
 #define RNG_CONFIG_MASK		(RNG_CR_ENTROPY_SRC_MASK | RNG_CR_CED | \
 				 RNG_CR_CLKDIV)
 
-#define RNG_MAX_NOISE_CLK_FREQ	U(3000000)
-
 struct stm32_rng_driver_data {
+	unsigned long max_noise_clk_freq;
 	bool has_power_optim;
 	bool has_cond_reset;
 	bool entropy_src_config;
@@ -273,7 +272,7 @@ static uint32_t stm32_rng_clock_freq_restrain(void)
 	 * No need to handle the case when clock-div > 0xF as it is physically
 	 * impossible
 	 */
-	while ((clock_rate >> clock_div) > RNG_MAX_NOISE_CLK_FREQ)
+	while ((clock_rate >> clock_div) > dev->ddata->max_noise_clk_freq)
 		clock_div++;
 
 	DMSG("RNG clk rate : %lu", clk_get_rate(dev->clock) >> clock_div);
@@ -665,6 +664,7 @@ err:
 
 static const struct stm32_rng_driver_data mp13_data[] = {
 	{
+		.max_noise_clk_freq = U(48000000),
 		.has_cond_reset = true,
 		.entropy_src_config = true,
 		.has_power_optim = true,
@@ -673,6 +673,7 @@ static const struct stm32_rng_driver_data mp13_data[] = {
 
 static const struct stm32_rng_driver_data mp15_data[] = {
 	{
+		.max_noise_clk_freq = U(48000000),
 		.has_cond_reset = false,
 		.entropy_src_config = false,
 		.has_power_optim = false,
@@ -682,6 +683,7 @@ DECLARE_KEEP_PAGER(mp15_data);
 
 static const struct stm32_rng_driver_data mp25_data[] = {
 	{
+		.max_noise_clk_freq = U(48000000),
 		.has_cond_reset = true,
 		.entropy_src_config = true,
 		.has_power_optim = true,
-- 
2.47.2

